### PR TITLE
Removing natural key fields attribute

### PR DESCRIPTION
**Pull request by daviddavis**
refs #3025
https://pulp.plan.io/issues/3025

### CODE REVIEW

File: platform/pulpcore/app/models/content.py [hunk 1]
```diff
@@ -90,22 +90,18 @@ class Content(MasterModel):
...
-    def natural_key(self):
-        """
-        Get the model's natural key based on natural_key_fields.
-
-        :return: The natural key.
-        :rtype: tuple
-        """
-        return tuple(getattr(self, f.name) for f in self.natural_key_fields)
+    @classmethod
+    def natural_key_fields(cls):
```
Comment (dkliban) [inline platform/pulpcore/app/models/content.py]: Want to add a docbloc here?

File: platform/pulpcore/app/models/content.py [hunk 2]
```diff
@@ -79,24 +79,28 @@ class Content(MasterModel):
...
     artifacts = models.ManyToManyField(Artifact, through='ContentArtifact')
 
     class Meta:
         verbose_name_plural = 'content'
+        unique_together = ()
+
+    @classmethod
+    def natural_key_fields(cls):
+        """
+        Returns a list of the natural key fields which usually equates to unique_together fields
```
Comment (dkliban) [inline platform/pulpcore/app/models/content.py]: This is a generator actually, so it yields field names as you iterate over the generator. If you want to convert it to return a list, I think that would be alright in this case. Either way, the docstring should match the behavior.

File: platform/pulpcore/app/models/content.py [hunk 3]
```diff
@@ -79,24 +79,26 @@ class Content(MasterModel):
...
+        """
+        return [field for fields in cls._meta.unique_together for field in fields]
```
Comment (jortel) [inline platform/pulpcore/app/models/content.py]: Did you consider using?: https://docs.python.org/3/library/itertools.html#itertools.chain.from_iterable
Like:
```
return list(chain.from_iterable(cls._meta.unique_together))
```

**Comment by jortel** [comment]
👍 I really like just doing this once the base class.

**Comment by daviddavis** [comment]
ok test

**Comment by daviddavis** [comment]
@jortel @dkliban I think this is ready to be reviewed again. Thanks for the reviews.

**Pull request merged**